docs/QSG/qemu-vms.md

```diff
@@ -0,0 +1,285 @@
+    COMMIT
+    # Completed on Fri Aug 24 15:20:25 2007
+    EOF
+    }
+    
+    start_dnsmasq() {
+        do_dnsmasq \
+        --strict-order \
+        --except-interface=lo \
+        --interface=$BRIDGE \
+        --listen-address=$GATEWAY \
+        --bind-interfaces \
+        --dhcp-range=$DHCPRANGE \
+        --conf-file="" \
+        --pid-file=/var/run/qemu-dnsmasq-$BRIDGE.pid \
+        --dhcp-leasefile=/var/run/qemu-dnsmasq-$BRIDGE.leases \
+        --dhcp-no-override \
+        ${TFTPROOT:+"--enable-tftp"} \
+        ${TFTPROOT:+"--tftp-root=$TFTPROOT"} \
+        ${BOOTP:+"--dhcp-boot=$BOOTP"}
+    }
+    
+    setup_bridge_nat() {
+        if check_bridge "$1" ; then
+        create_bridge "$1"
+        enable_ip_forward
+        add_filter_rules "$1"
+        start_dnsmasq "$1"
+        fi
+    }
+    
+    setup_bridge_vlan() {
+        if check_bridge "$1" ; then
+        create_bridge "$1"
+        start_dnsmasq "$1"
+        fi
+    }
+    
+    setup_bridge_nat "$BRIDGE"
+    
+    if test "$1" ; then
+        do_ifconfig "$1" 0.0.0.0 up
+        do_brctl addif "$BRIDGE" "$1"
+    fi
+```
+Then add a sudo rule for qemu-system-x86<sub>64</sub> in *etc/sudoers.d*.
+```
+    <username> blackbox=(root) NOPASSWD: /usr/bin/qemu-system-x86_64
+```
+3.  Install guest operating systems.
+
+Run these two commands. Select "boot from cdrom". Follow the installation guide to install the OSes.
+```
+    sudo qemu-system-x86_64 -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 3 -m 12288 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-server.qcow2,if=virtio -device virtio-net,netdev=mynet0,mac=52:54:00:12:34:56 -netdev tap,id=mynet0 -drive file=<image-dir>/qemu-nvm-disk1.qcow2,if=none,id=nvm1 -device nvme,serial=deadbeef,drive=nvm1 -cdrom <iso-dir>/Rocky-8.9-x86_64-minimal.iso -boot menu=on &
+    sudo qemu-system-x86_64  -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 1 -m 2048 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-client.qcow2,if=virtio -device virtio-net,netdev=mynet1,mac=52:54:00:12:34:57 -netdev tap,id=mynet1 -cdrom <iso-dir>/Rocky-8.9-x86_64-minimal.iso -boot menu=on
+```
+4.  Install DAOS software.
+
+After OS installation is finished. Remove the `-cdrom` and `-boot` options from the QEMU commands to boot VMs normally. Next is to install DAOS software.
+```
+    sudo qemu-system-x86_64 -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 3 -m 12288 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-server.qcow2,if=virtio -device virtio-net,netdev=mynet0,mac=52:54:00:12:34:56 -netdev tap,id=mynet0 -drive file=<image-dir>/qemu-nvm-disk1.qcow2,if=none,id=nvm1 -device nvme,serial=deadbeef,drive=nvm1 &
+    sudo qemu-system-x86_64  -M q35,accel=kvm,kernel-irqchip=split -cpu host -smp 1 -m 2048 -device intel-iommu,intremap=on -drive file=<image-dir>/daos-client.qcow2,if=virtio -device virtio-net,netdev=mynet1,mac=52:54:00:12:34:57 -netdev tap,id=mynet1 
+```
```

Comment:
enable the interface once the VM is booted 
![image](https://github.com/user-attachments/assets/98b3a271-91d7-4b61-b437-c1b5e1f18595)

Reply:
nmcli c modify enp0s2 connection.autoconnect "yes"
use 
nmcli c show enp9s2 
to confirm

Reply:
done